pkg/utils/apply/apply.go

```diff
@@ -0,0 +1,205 @@
+package apply
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// An Object is a Kubernetes object.
+type Object interface {
+	metav1.Object
+	runtime.Object
+}
+
+// Applicator provides two approaches to applies changes to an object,
+// patching and updating.
```

Comment:
We can provide only PatchingApplicator here, then it's clear.

If we provide both patch and update. We must explain more details for the difference of these two approaches. Or it may cause lot's of confusion.

Reply:
Totally agreed. I think we don't need updatingApply anymore. Let me remove it.